presentation/src/main/java/com/architect/coders/mu8/comics/ComicsViewModel.kt

```diff
@@ -0,0 +1,43 @@
+package com.architect.coders.mu8.comics
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import com.architect.coders.mu8.comics.ComicsViewModel.UiModel.*
+import com.architect.coders.mu8.utils.ScopedViewModel
+import com.architect.codes.mu8.interactor.comics.ComicUseCase
+import com.architect.codes.mu8.model.Comic
+import kotlinx.coroutines.launch
+
+class ComicsViewModel(private val useCase: ComicUseCase) : ScopedViewModel() {
+
+    sealed class UiModel {
+        object ShowLoading : UiModel()
+        object InternetError : UiModel()
+        class LoadData(val comics: List<Comic>) : UiModel()
+        class NavigateTo(val cominc: Comic) : UiModel()
+    }
+
+    private val _model = MutableLiveData<UiModel>()
+
+    val model: LiveData<UiModel>
+        get() {
+            if (_model.value == null) loadData()
+            return _model
+        }
+
+    private fun loadData() {
+        launch {
+            _model.value = ShowLoading
+
+            try {
+                _model.value = LoadData(useCase.getComics())
+            } catch (e: Exception) {
```

Comment:
Please don't catch a generic exception, catch an specific one like the IO or any other you need.

Reply:
At the moment I'm not sure the exception kind.